src/test/java/uk/ac/ebi/ega/egafuse/service/EgaFileServiceTest.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.service;
+
+import static okhttp3.mock.Behavior.UNORDERED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+
+import jnr.ffi.Pointer;
+import okhttp3.OkHttpClient;
+import okhttp3.mock.MockInterceptor;
+import uk.ac.ebi.ega.egafuse.config.EgaFuseApplicationConfig;
+import uk.ac.ebi.ega.egafuse.model.CacheKey;
+import uk.ac.ebi.ega.egafuse.model.File;
+
+@TestPropertySource("classpath:application-test.properties")
+@ContextConfiguration(classes = EgaFuseApplicationConfig.class)
+@RunWith(SpringRunner.class)
+public class EgaFileServiceTest {
+    private EgaFileService egaFileService;
+    private MockInterceptor interceptor;
+    private OkHttpClient client;
+    private ObjectMapper objectMapper;
+
+    @Value("${api.chunksize}")
+    private long CHUNK_SIZE;
+
+    @Value("${cache.prefetch}")
+    private int PREFETCH;
+
+    @Value("${app.url}")
+    private String APP_URL;
+
+    @Mock
+    private AsyncLoadingCache<CacheKey, byte[]> cache;
+
+    @Mock
+    private Token token;
+
+    @Mock
+    private Pointer pointer;
+
+    @Before
+    public void before() {
+        objectMapper = new ObjectMapper();
+        interceptor = new MockInterceptor(UNORDERED);
+        client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+        egaFileService = new EgaFileService(client, APP_URL, CHUNK_SIZE, PREFETCH, token, cache);
+    }
+
+    @Test
+    public void testGetFiles() throws JsonProcessingException {
```

Comment:
Could you please make these test names more specific please? It's in the whole file, it's important to write what they are actually testing. Please have a look on Ashutosh examples they are very easy to understand what the test is doing.